packages/contracts-bedrock/scripts/Deployer.sol

```diff
@@ -561,15 +563,44 @@ abstract contract Deployer is Script {
     }
 
     /// @dev Returns the value of the internal `_initialized` storage slot for a given contract.
-    function loadInitializedSlot(string memory _contractName, bool _isProxy) public returns (uint8 initialized_) {
-        StorageSlot memory slot = getInitializedSlot(_contractName);
-        if (_isProxy) {
-            _contractName = string.concat(_contractName, "Proxy");
+    function loadInitializedSlot(string memory _contractName) public returns (uint8 initialized_) {
+        uint256 contractNameLength = bytes(_contractName).length;
+        address addr;
+        // Check if the contract name ends with `Proxy` and if so, get the implementation address
+        string memory suffix = _substring(_contractName, contractNameLength - 5, contractNameLength);
+        if (bytes(suffix).length == bytes("Proxy").length && keccak256(bytes(suffix)) == keccak256(bytes("Proxy"))) {
+            addr = EIP1967Helper.getImplementation(getAddress(_contractName));
+            _contractName = _substring(_contractName, 0, contractNameLength - 5);
+            // If the EIP1967 implementation address is 0, we try to get the implementation address from legacy
+            // AddressManager, which would work if the proxy is ResolvedDelegateProxy like L1CrossDomainMessengerProxy.
+            if (addr == address(0)) {
+                addr = AddressManager(mustGetAddress("AddressManager")).getAddress(string.concat("OVM_", _contractName));
+            }
+        } else {
+            addr = mustGetAddress(_contractName);
         }
-        bytes32 slotVal = vm.load(mustGetAddress(_contractName), bytes32(vm.parseUint(slot.slot)));
+        StorageSlot memory slot = getInitializedSlot(_contractName);
+        bytes32 slotVal = vm.load(addr, bytes32(vm.parseUint(slot.slot)));
         initialized_ = uint8((uint256(slotVal) >> (slot.offset * 8)) & 0xFF);
     }
 
+    function _substring(
```

Comment:
I'm surprised we don't already have this functionality somewhere. 
@mds1 I looked in forge-std, but couldn't find it there, did I miss something? 

I think it'd be worth either: 
1. starting a new file: `src/libraries/Strings.sol`
2. contributing this to `forge-std`

Reply:
I agree on using solady here, we should add it as a dep and then we can remove our vendor of solady sort

Reply:
I think it is a dep?

Reply:
(just unresolved this comment so it doesn't get lost, since I didn't see this change introduced yet)

Reply:
had locally made the commit but didn't push it. plz check now!

Reply:
Looks good now, thanks! Only comment would be to pin to the latest tag, (`v0.0.158`, commit 502cc1e) instead of the most recent commit (currently using ac6d2cc). 

You can check the current version we're using with `git submodule status`, and can install a tag with `forge install https://github.com/Vectorized/solady@v0.0.158` (though I thought `forge install https://github.com/Vectorized/solady` would do this by default)

Reply:
good idea. thank you!